### PR TITLE
VACMS-6445: Make product field required.

### LINKED ISSUE #6445: Make product field required on KB article
## User Story or Problem Statement

As a KB article editor, i should be required to choose a product. 

## Acceptance Criteria
- [ ] Products field is required on KB article.

## Implementation steps
<optional>

## Design principles

Which [CMS design principle](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/cms/cms-product-design-principles.md) is at play?
- [ ] Purpose-driven: Creates an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] Efficient: Removes distractions and create clear, straightforward paths to get the job done.
- [ ] Approachable: Offers friendly guidance over authoritative instruction.
- [x] Consistent: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] Empowering: Provide clear information to help editors make decisions about their work.

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

closes #6445 

## Testing done
Created new CMS KB article locally and saw the products field was required.

## Screenshots
image

## QA steps

As a cms editor, I can edit or create a new KB article and am required to choose a product.

1. Then validate Acceptance Criteria from issue
- [x] Products field is required on KB article.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change.
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by kevwalsh — CHANGES_REQUESTED** [review]
Tests are still running, but i'm sure Behat test will fail. You'll need to update the drupal spec tool and then copy/paste the behat test it generates. [Instructions are here](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/master/READMES/workflow.md#drupal-spectool).

**Comment by swirtSJW** [comment]
Test wise:  The behat fail makes sense.  The accessibility does not I'll look into it after the behat is updated.

**Comment by swirtSJW** [comment]
composer.json needs a 
```
"3213473 - Disallow aria-required on  checkbox.": "https://www.drupal.org/files/issues/2021-06-04/3213473-27.patch",
```

and hopefully that will get tests passing.

**Comment by swirtSJW** [comment]
I found part of the problem on the drupal.org issue.
https://www.drupal.org/project/drupal/issues/3213473

Somebody misnumbered the patch.   patch 27 comes before patch 24  :)    Patch 24 is the one we should be using.

**Comment by ethanteague** [comment]
Just letting you all know I'm aware of the last fail and working on it.

File: tests/behat/drupal-spec-tool/content_model_user_guides_content_type_fields.feature [hunk 1]
```diff
@@ -1,17 +1,17 @@
-@api
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_user_guides_content_type_fields.feature:1]: @msbtterswrth  his looks like that line ending issue again.  This should only be showing a change to line 17.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
All looks good except for the line ending jumping in the behat feature.

File: tests/behat/drupal-spec-tool/content_model_user_guides_content_type_fields.feature (hunk 1 above)
Comment (ethanteague) [reply tests/behat/drupal-spec-tool/content_model_user_guides_content_type_fields.feature:1]: This is strange. While checked out on this branch, I pulled down the version of this file that lives on `main` and was still seeing a diff against main showing everything as being moved, @swirtSJW

Comment (msbtterswrth) [reply tests/behat/drupal-spec-tool/content_model_user_guides_content_type_fields.feature]: Updated!

**Review by swirtSJW — APPROVED** [review]

**Pull request merged**